### ISSUE TITLE
Installing gnome, gdm and gnome-tweaks fails

### ISSUE THREAD

**Issue by Cornul11** [archlinux/archinstall#1068]
Choosing gnome as the desktop environment will fail at one of the last stages if the device's hardware clock or system clock is wrong or the timezone is set wrongfully. This leads to GPG keys failing to be imported or validated during the integrity checks, which leads to a failure.
I would argue that it would be great to enforce some time sync of the hardware/system clock and an NTP synchronization before the package installation phase.

**Comment by Torxed** [comment]
We do, but it's optional as we wish to enforce as little as possible on the users. Selecting NTP during the first stages of the installer will perform a NTP sync before installing packages :)

**Comment by Cornul11** [comment]
I did choose to perform an NTP sync earlier in the installation process, but the problem persisted up until I forcefully synced manually from outside the archinstall script.

> We do, but it's optional as we wish to enforce as little as possible on the users. Selecting NTP during the first stages of the installer will perform a NTP sync before installing packages :)

**Comment by Torxed** [comment]
That's strange, I'll have a try here.
It might be that the time sync took too long and you ran through the menue's quick enough? We don't perform a check currently if the NTP service has finished syncing up (which can take anywhere between 0.5 seconds - 1 minute if the latency is bad).

I'll leave this open and push a check for the NTP service prior to running `minimal_installation()` if NTP has been chosen!

**Comment by Torxed** [comment]
I'm having trouble recreating this issue.
This is for me the Archiso releng just clean build with archinstall autostarting:
![screenshot](https://user-images.githubusercontent.com/861439/162943183-a1c5f706-4342-4601-8450-c0dd744ab261.jpg)

To me this is a behavior we need to disable, the fact that it *forcefully* actives NTP without asking the users for it.
So if anything, this issue has shed light on this behavior.. But I can't recreated time being out of sync other than the sync not having access to network while trying to perform the NTP sync. (not sure how often it retries)

**Comment by Cornul11** [comment]
I am having trouble recreating the issue with the latest `archiso` as well. I had a maybe two-month-old `archiso` when I originally encountered the issues. It might have been the case that the hardware clock was somehow messed up by the Windows time sync, which I know does not use UTC.

I just used the script and it worked flawlessly for me so probably the issue is not relevant anymore.

**Comment by Cornul11** [comment]
I am sure that the Archiso I had was asking for it, so NTP sync was already a thing in the archinstall version bundled with the Archiso I used. So it may be the case that there is a different reason for the NTP not working correctly due to the dual-boot present on the system. Anyway, that's all speculation as long as the problem is reproducible. Thank you, nevertheless, for a great script and tool!

**Comment by Torxed** [comment]
This should have been fixed in `master` (v2.4.0)  now.
I've tweaked it so that NTP is only activated if selected through the menu in commit 9a6e85fb4663c4cabf2a4ce573fc9263a291f2e2

If you select to activate NTP (for the installed system), it will also run `timedatectl set-ntp true`.
Finally, I merged a PR that only lets through to the installation if a time-sync has been established with at least one server. That was done in PR #1086
